jina/types/arrays/document.py

```diff
@@ -307,7 +307,13 @@ def save_binary(self, file: Union[str, BinaryIO]) -> None:
 
         with file_ctx as fp:
             dap = DocumentArrayProto()
-            dap.docs.extend([d.proto for d in self._docs_proto])
+            if len(self._docs_proto) > 0:
+                if type(self._docs_proto[0]) != DocumentProto:
+                    dap.docs.extend([d.proto for d in self._docs_proto])
+                else:
+                    dap.docs.extend([d for d in self._docs_proto])
+            else:
+                default_logger.warning('data is empty')
```

Comment:
do not raise error or warning, empty content is a valid content in IO ops

Reply:
Sure